src/test/java/com/atlassian/db/replica/api/AuroraClusterDiscoveryTest.java

```diff
@@ -46,9 +46,12 @@ public void before() throws SQLException {
             auroraClusterMock.getMainConnection(),
             "jdbc:postgresql://" + readerEndpoint + "/" + databaseName
         );
-        auroraClusterDiscovery = new AuroraClusterDiscovery(
-            anAuroraConnectionDetailsBuilder().username(jdbcUsername).password(jdbcPassword).build()
-        );
+        auroraClusterDiscovery = new AuroraClusterDiscovery((url) -> () -> {
+            Properties properties = new Properties();
+            properties.setProperty("user", jdbcUsername);
+            properties.setProperty("password", jdbcPassword);
+            return DriverManager.getConnection(url, properties);
+        });
```

Comment:
simple usage of new spi interfaces. creates and returns a new connection every time it is requested